### ISSUE TITLE
about data split

### ISSUE THREAD

**Issue by yangzhao1230** [Liuzhe30/EMO#1]
Hi, 

Thanks for your great work and well-organized GitHub repo!

I have a question about data split. You mention an "8:1:1" ratio for data split. But you only provide a train split and a test split for each size data. 

<img width="723" height="269" alt="Image" src="https://github.com/user-attachments/assets/fd28101a-fe4e-4b74-9a52-169c2c8bc099" />

I'm wondering how to get validation data?

Thanks!

**Comment by Liuzhe30** [comment]
Thanks a lot for your interest in our work and for checking out the repo!

Regarding your question about the 8:1:1 data split, the validation set is created directly from the training split during training. You can find the relevant lines here in https://github.com/Liuzhe30/EMO/blob/722e3f5a867951204afc9ad9457cdb9ab910d962/training.py#L52-L53

Depending on your purpose:
1. If you want to evaluate the model’s fitting performance, you can shuffle the training set first and then re-split a validation subset by ratio.
2. If you plan to use our pretrained model for prediction or fine-tuning, you can directly use the released model weights we provided in the repo.

Hope this helps!

Best,
Zhe

**Comment by yangzhao1230** [comment]
Thanks for your quick reponse.

The code appears to directly use the last 1/9 of the training split as the validation set, so I'm wondering why your first piece of advice suggests shuffling the training set and then re-splitting it to create a validation set?

**Comment by Liuzhe30** [comment]
Yes, you’re right. In our experiments we used the last 1/9 of the training data as the validation set. If you are reproducing our work, please use the same split.

I mentioned shuffling only as an optional step to check whether the model performance remains stable when the data order changes (if you need).

Hope that clears it up!

**Comment by yangzhao1230** [comment]
Thank you! My question has been successfully solved.
